software/controller/lib/hal/stepper.cpp

```diff
@@ -395,113 +347,90 @@ StepMtrErr StepMotor::SetAmpAll(float amp) {
 }
 
 // Immediately disable the motor
-StepMtrErr StepMotor::HardDisable() {
-  uint8_t cmd = static_cast<uint8_t>(StepMtrCmd::HardDisable);
-  return SendCmd(&cmd, 1);
+ErrorCode Handler::HardDisable() {
+  uint8_t command = static_cast<uint8_t>(OpCode::HardDisable);
+  return SendCmd(&command, 1);
 }
 
 // Reset the motor position to zero
-StepMtrErr StepMotor::ClearPosition() {
-  uint8_t cmd = static_cast<uint8_t>(StepMtrCmd::ResetPosition);
-  return SendCmd(&cmd, 1);
+ErrorCode Handler::ClearPosition() {
+  uint8_t command = static_cast<uint8_t>(OpCode::ResetPosition);
+  return SendCmd(&command, 1);
 }
 
 // Reset the stepper chip
-StepMtrErr StepMotor::Reset() {
-  uint8_t cmd = static_cast<uint8_t>(StepMtrCmd::ResetDevice);
-  return SendCmd(&cmd, 1);
+ErrorCode Handler::Reset() {
+  uint8_t command = static_cast<uint8_t>(OpCode::ResetDevice);
+  return SendCmd(&command, 1);
 }
 
-StepMtrErr StepMotor::GetStatus(StepperStatus *stat) {
-  // It's not legal to call this from the control loop because it
-  // has to block.  Return an error if we're in an interrupt handler
+ErrorCode Handler::GetStatus(Status *stat) {
+  // It's not legal to call this from the control loop because it has to block.
+  // Return an error if we're in an interrupt handler
   if (Interrupts::singleton().InInterruptHandler()) {
-    return StepMtrErr::WouldBlock;
+    return ErrorCode::WouldBlock;
   }
 
   // We expect a 2 bytes response so we add two Nops after the op code.
-  uint8_t cmd[] = {static_cast<uint8_t>(StepMtrCmd::GetStatus),
-                   static_cast<uint8_t>(StepMtrCmd::Nop), static_cast<uint8_t>(StepMtrCmd::Nop)};
-  uint8_t response[sizeof(cmd) - 1] = {0};
+  uint8_t command[] = {static_cast<uint8_t>(OpCode::GetStatus), static_cast<uint8_t>(OpCode::Nop),
+                       static_cast<uint8_t>(OpCode::Nop)};
+  uint8_t response[sizeof(command) - 1] = {0};
 
-  StepMtrErr err = SendCmd(cmd, sizeof(cmd), response);
-  if (err != StepMtrErr::Ok) {
+  ErrorCode err = SendCmd(command, sizeof(command), response);
+  if (err != ErrorCode::Ok) {
     return err;
   }
 
-  stat->enabled = (response[1] & 0x01) == 0;
-  stat->command_error = (response[1] & 0x80) || (response[0] & 0x01);
-  stat->under_voltage = (response[0] & 0x02) == 0;
-  stat->thermal_warning = (response[0] & 0x04) == 0;
-  stat->thermal_shutdown = (response[0] & 0x08) == 0;
-  stat->over_current = (response[0] & 0x10) == 0;
-  stat->step_loss = (response[0] & 0x60) != 0x60;
-
-  switch (response[1] & 0x60) {
-    case 0x00:
-      stat->move_status = StepMoveStatus::Stopped;
-      break;
-    case 0x20:
-      stat->move_status = StepMoveStatus::Accelerating;
-      break;
-    case 0x40:
-      stat->move_status = StepMoveStatus::Decelerating;
-      break;
-    case 0x60:
-      stat->move_status = StepMoveStatus::ConstantSpeed;
-      break;
-  }
+  // received MSB first
+  stat->word = static_cast<uint16_t>((response[0] << 8) + response[1]);
```

Comment:
So, I assume the more intricate conversion is no longer necessary because you have mapped that word out in the new union up top?